enterprise/server/auth/auth.go

```diff
@@ -252,17 +259,84 @@ func (a *authenticator) verifyTokenAndExtractUser(ctx context.Context, jwt strin
 	return extractToken(a.issuer, validToken)
 }
 
+type apiKeyGroupCacheEntry struct {
+	data         interfaces.APIKeyGroup
+	expiresAfter time.Time
+}
+
+// apiKeyGroupCache is a cache for API Key -> Group lookups. A single Bazel invocation
+// can generate large bursts of RPCs, each of which needs to be authed.
+// There's no need to go to the database for every single request as this data
+// rarely changes.
+type apiKeyGroupCache struct {
+	// Note that even though we base this off an LRU cache, every entry has a hard expiration
+	// time to force a refresh of the underlying data.
+	lru *lru.LRU
+	ttl time.Duration
+	mu  sync.RWMutex
+}
+
+func newAPIKeyGroupCache(configurator *config.Configurator) (*apiKeyGroupCache, error) {
+	var err error
+	var ttl time.Duration
+	if configurator.GetAuthAPIKeyGroupCacheTTL() != "" {
+		ttl, err = time.ParseDuration(configurator.GetAuthAPIKeyGroupCacheTTL())
+		if err != nil {
+			return nil, fmt.Errorf("invalid API Key -> Group cache TTL [%s]: %v", configurator.GetAuthAPIKeyGroupCacheTTL(), err)
+		}
+	} else {
+		ttl = defaultAPIKeyGroupCacheTTL
+	}
+
+	sizeFunc := func(k, v interface{}) int64 { return 1 }
```

Comment:
I think this is fine as-is -- I expect this to be really small, but if you wanted to guarantee the auth cache was less than 10MB or something -- defining a real sizeFunc might be better.

Reply:
This cache will be tiny so I don't think we need to get any fancier.